### ISSUE TITLE
Split package

### ISSUE THREAD

**Issue by jkarni** [haskell-servant/servant-multipart#15]
`servant-multipart`, unlike most packages which introduce new combinators, isn't split up (into `-server`, `-docs`, etc.). So any API that uses it will incur all depedencies (and likely can't run in GHCJS, etc.).

@alpmestan @phadej I imagine I can go ahead and split it, no?

**Comment by alpmestan** [comment]
@jkarni Absolutely! I didn't bother initially because it just had the server interpretation, then we got -docs support and I was planning to do the split as soon as we get -client support. If you want to do it now, be my guest!

(Someone in fact contacted me today, about implementing -client support)
